components/ipfs/ipfs_service.cc

```diff
@@ -371,6 +376,91 @@ void IpfsService::NotifyIpnsKeysLoaded(bool result) {
   }
 }
 
+// Local pinning
+void IpfsService::AddPin(const std::vector<std::string>& cids,
+                         bool recursive,
+                         AddPinCallback callback) {
+  if (!IsDaemonLaunched()) {
+    std::move(callback).Run(false, absl::nullopt);
+    return;
+  }
+
+  GURL gurl = server_endpoint_.Resolve(kAddPinPath);
+  for (const auto& cid : cids) {
+    gurl = net::AppendQueryParameter(gurl, kArgQueryParam, cid);
+  }
+  gurl = net::AppendQueryParameter(gurl, "recursive",
+                                   recursive ? "true" : "false");
+
+  auto url_loader = std::make_unique<api_request_helper::APIRequestHelper>(
```

Comment:
That's super weird. Why we create APIRequestHelper on each request? There is supposed to be only one instance of it per class.

Reply:
Ok, this class went wrong way of using APIRequestHelper a while ago in this PR https://github.com/brave/brave-core/pull/15070/files#diff-3c9e89e7e5f2725cc6df997848bc45097b1602d21cf68d255a11e04414f583a5

Can you please put a TODO somewhere to refactor this?

Reply:
https://github.com/brave/brave-browser/issues/27478